src/app/OrderButton.tsx

```diff
@@ -95,21 +95,16 @@ export function OrderButton() {
       account &&
       gatewayApi
     ) {
+      // TODO: Move all logic to separate pure TS class/classes/file/files.
+      // Context providers will recieve only the data necessary to render the page.
+      // This will allow cleaner code and better incapsulation.
       fetchBalances(
         adexState.currentPairInfo.token1.address,
         adexState.currentPairInfo.token2.address,
         account
       );
     }
-  }, [
-    adexState.currentPairInfo.token1.address,
-    adexState.currentPairInfo.token2.address,
```

Comment:
Why this fixes the bug?
These addresses don't change when we make a transaction. `fetchBalances` doesn't get called every time when `adexState` is updated, only when we change addresses. So [the subscription event](https://github.com/DeXter-on-Radix/website/blob/fix-balance-update/src/app/layout.tsx#L81) gets ignored.